pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExporter.java

```diff
@@ -47,20 +56,67 @@
 
     // A cache for user auth0 emails, storing (auth0UserId -> email).
     private static final Map<String, String> emailStore = new HashMap<>();
+    private static final String familyIdHeader = "familyId";
 
-    private final Config cfg;
+    private final Config mainConfig;
+    private final Config exportConfig;
     private final String customGuid;
     private final String customActivity;
 
-    public CustomExporter(Config cfg) {
-        this.cfg = cfg;
-        this.customActivity = cfg.getString(CustomExportConfigFile.ACTIVITY);
-        this.customGuid = cfg.getString(CustomExportConfigFile.STUDY_GUID);
+    public CustomExporter(Config mainConfig, Config exportConfig) {
+        this.mainConfig = mainConfig;
+        this.exportConfig = exportConfig;
+        this.customActivity = exportConfig.getString(CustomExportConfigFile.ACTIVITY);
+        this.customGuid = exportConfig.getString(CustomExportConfigFile.STUDY_GUID);
     }
 
+    List<CustomExportParticipant> extractParticipantDataSetByIds(Handle handle, StudyDto studyDto, Set<Long> userIds)
+            throws IOException {
+        RestHighLevelClient elasticsearchClient = ElasticsearchServiceUtil.getElasticsearchClient(this.mainConfig);
+        return createCustomParticipantsFrom(ExportUtil.extractParticipantDataSetByIds(handle, studyDto, userIds, emailStore), handle,
+                studyDto, elasticsearchClient);
+    }
+
+    private static List<CustomExportParticipant> createCustomParticipantsFrom(List<Participant> participants, Handle handle,
+                                                                              StudyDto studyDto,
+                                                                              RestHighLevelClient esClient) {
+        List<CustomExportParticipant> customExportParticipants = new ArrayList<>();
+        for (Participant p : participants) {
+            String familyId = getFamilyId(handle, studyDto, p.getUser().getGuid(), esClient);
```

Comment:
I was going to suggest something more efficient, like a bulk query or something, but thinking about the use-case here, I see that most of the time this is an incremental export with a handful of new participants, so I guess this is okay.

Reply:
I tried doing this with a boolean search query and ElasticSearch complained about having to sort the results so I think I'll keep it this way for now.  Good point, though.